src/components/ControlInput.vue

```diff
@@ -133,6 +134,10 @@ export default {
                 this.active_tool = tool;
             }
         },
+        //Check the users settings to see if they have turned on the colour picker
+        displayColourPicker: function displayColourPicker() {
+           this.display_colour_picker = state.setting('showColourPicker');
+        },
```

Comment:
This `displayColourPicker()` function would be better suited in the computed section of the component and the `display_colour_picker` data property removed. eg:
~~~javascript
shouldDisplayColourPicker() {
    return state.setting('showColourPicker');
}
~~~

You would then use it like `v-if="shouldDisplayColourPicker"`